src/Uno.UI/UI/Xaml/Window/Native/NativeWindowWrapper.Android.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using Android.App;
+using Android.Runtime;
+using Android.Util;
+using Android.Views;
+using AndroidX.Core.View;
+using Uno.UI.Extensions;
+using Windows.ApplicationModel.Core;
+using Windows.Foundation;
+using Windows.UI.Core;
+using Windows.UI.ViewManagement;
+using Size = Windows.Foundation.Size;
+
+namespace Uno.UI.Xaml.Controls;
+
+internal class NativeWindowWrapper : NativeWindowWrapperBase
+{
+	private static readonly Lazy<NativeWindowWrapper> _instance = new(() => new NativeWindowWrapper());
+
+	private readonly ActivationPreDrawListener _preDrawListener;
+	private Rect _previousTrueVisibleBounds;
+
+	public NativeWindowWrapper()
+	{
+		_preDrawListener = new ActivationPreDrawListener(this);
+		CoreApplication.GetCurrentView().TitleBar.ExtendViewIntoTitleBarChanged += RaiseNativeSizeChanged;
+	}
+
+	public override object NativeWindow => Windows.UI.Xaml.ApplicationActivity.Instance?.Window;
+
+	internal static NativeWindowWrapper Instance => _instance.Value;
+
+	internal int SystemUiVisibility { get; set; }
+
+	internal void OnNativeVisibilityChanged(bool visible) => Visible = visible;
+
+	internal void OnActivityCreated() => AddPreDrawListener();
+
+	internal void OnNativeActivated(CoreWindowActivationState state) => ActivationState = state;
+
+	internal void OnNativeClosed() => RaiseClosed();
+
+	internal bool IsStatusBarTranslucent()
+	{
+		if (!(ContextHelper.Current is Activity activity))
+		{
+			throw new Exception("Cannot check NavigationBar translucent property. Activity is not defined yet.");
+		}
+
+		return activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.TranslucentStatus)
+			|| activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.LayoutNoLimits);
+	}
+
+	internal void RaiseNativeSizeChanged()
+	{
+		var (windowSize, visibleBounds, trueVisibleBounds) = GetVisualBounds();
+
+		Bounds = new Rect(default, windowSize);
+		VisibleBounds = visibleBounds;
+
+		if (_previousTrueVisibleBounds != trueVisibleBounds)
+		{
+			_previousTrueVisibleBounds = trueVisibleBounds;
+
+			// TODO: Adjust when multiple windows are supported on Android #13827
+			ApplicationView.GetForCurrentView()?.SetTrueVisibleBounds(trueVisibleBounds);
+		}
+	}
+
+	protected override void ShowCore() => RemovePreDrawListener();
+
+	private (Size windowSize, Rect visibleBounds, Rect trueVisibleBounds) GetVisualBounds()
+	{
+		if (ContextHelper.Current is not Activity activity)
+		{
+			return default;
+		}
+
+		var windowInsets = ViewCompat.GetRootWindowInsets(activity.Window.DecorView);
+
+		var insetsTypes = WindowInsetsCompat.Type.SystemBars(); // == WindowInsets.Type.StatusBars() | WindowInsets.Type.NavigationBars() | WindowInsets.Type.CaptionBar();
```

Comment:
Remove or useful ?

Reply:
This is just copied from what was in Window.Android.cs, so I would keep it. I think it just explains what the method does